### ISSUE TITLE
alert rule editable permission is not working as expected

### ISSUE THREAD

**Issue by bossjie** [grafana/grafana#65767]
**What happened**:
Editor could edit&save changes for an unallowed alert rule.

**What you expected to happen**:
editor is not expected to be able to do this as I set it up in alert rule's folder

**How to reproduce it (as minimally and precisely as possible)**:
1.login using userA(role: editor)  create a folder named "testAccess" and set permission  Editor can View
2.create alert rule named "alertName1" and save it into "testAccess"
3.login using userB(role: editor), go to alert rules tab and edit above alert rule, change its name from  "alertName1" to  "alertName2"  click save button 
4. the alert name is successfully changed to  "alertName2"

**Anything else we need to know?**:
when I set the folder permission,  remove Editor Can View 
userB as an editor, does cannot see target alert rule.  which means it works for this case but editable case as above, it doesn't work even I turn on and turn off for an option in config:  editors_can_admin = false/true

**Environment**: centos 8
- Grafana version: 9.4.7
- Data source type & version: NA
- OS Grafana is installed on: centos 8
- User OS & Browser: mac, chrome
- Grafana plugins: NA
- Others: NA

**Comment by mellieA** [comment]
I confirmed this behavior with the steps: 

Create a folder as admin, edit permissions to give the Editor role "view" permissions only
Create a user that has Editor role
Create an alert and put it in the restricted folder
Log in as your test user
Navigate to the test alert, edit the alert, click save - the alert is saved

Additional test - 

As Editor, remove the editor permission to the folder
Added User permission for the test user as "viewer" role with view permission
Create a dashboard in the test folder
Log in as the test user
Navigate to the test alert, edit the alert, click save - the alert is saved in both scenarios when given the "View" permission
Navigate to a dashboard in the same folder - there are no edit permissions

**Comment by bossjie** [comment]
correct,  dashboard editable/viewable permission works while the alert rule not.

**Comment by bossjie** [comment]
Hi Team, do we have any update on it?  thanks in advance.

**Comment by bossjie** [comment]
Hi Team, do we have any update on it? thanks in advance.
we met this issue and have no idea to go over.  @armandgrillet
